utils/node_local_test.py

```diff
@@ -6350,25 +6489,51 @@ def run(wf, args):
                         fatal_errors=fatal_errors) as server:
             if args.mode == 'launch':
                 run_in_fg(server, conf, args)
-            elif args.mode == 'overlay':
+            elif args.mode == 'overlay' and 'special_dfuse_overlay' in special_list:
                 fatal_errors.add_result(run_duns_overlay_test(server, conf))
             elif args.mode == 'set-fi':
                 fatal_errors.add_result(server.set_fi())
             elif args.mode == 'all':
                 fi_test_dfuse = True
-                fatal_errors.add_result(run_posix_tests(server, conf))
-                fatal_errors.add_result(run_dfuse(server, conf))
-                fatal_errors.add_result(run_duns_overlay_test(server, conf))
+                fatal_errors.add_result(run_posix_tests(server, conf, test_dict.keys()))
+                if 'special_dfuse_multi' in special_list:
+                    fatal_errors.add_result(run_dfuse(server, conf))
+                if 'special_dfuse_overlay' in special_list:
+                    fatal_errors.add_result(run_duns_overlay_test(server, conf))
                 test_pydaos_kv(server, conf)
                 test_pydaos_kv_obj_class(server, conf)
                 fatal_errors.add_result(server.set_fi())
             elif args.test == 'all':
-                fatal_errors.add_result(run_posix_tests(server, conf))
+                fatal_errors.add_result(run_posix_tests(server, conf, test_dict.keys()))
             elif args.test:
-                fatal_errors.add_result(run_posix_tests(server, conf, args.test))
+                special_list = [x for x in args.test if is_special_testname(x)]
+                despecialed_list = ['test_' + x for x in args.test if not is_special_testname(x)]
+                custom_test_dict = expand_input_list(despecialed_list)
+                custom_exclusions = explicit_list_to_exclusion_list(custom_test_dict)
+                exclusion_union = {}
+                for key in custom_test_dict.keys():
+                    exclusion_list = \
+                        list(set(custom_exclusions[key]
+                                 if key in custom_exclusions else []).union(
+                                     set(excluded_dict[key] if key in excluded_dict else [])))
```

Comment:
FYI `dict.get` doesn't raise an exception and takes a default value. E.g.
```
>>> {}.get('foo', 'some_default')
'some_default'
```

So you could do this
```suggestion
                        list(set(custom_exclusions.get(key, []).union(
                                 set(excluded_dict.get(key, [])))
```

Reply:
Fixed.